panel/param.py

```diff
@@ -817,13 +821,17 @@ def _validate_object(self):
             "class or global state in an application's namespace."
         )
 
+    @classmethod
+    def applies(cls, obj: Any) -> float | bool | None:
+        return bool(resolve_ref(obj))
+
     #----------------------------------------------------------------
     # Callback API
     #----------------------------------------------------------------
 
     @classmethod
-    def eval(self, function):
-        return eval_function_with_deps(function)
+    def eval(self, ref):
```

Comment:
```suggestion
    def eval(cls, ref):
```
As you are already making some changes in this area